### ISSUE TITLE
Lists Improvements

### ISSUE THREAD

**Issue by DebbieSan** [internetarchive/openlibrary#7390]
It is disappointing that to create a list, one needs to navigate through many pages. Adding a list is not intuitive and many users may not create lists because they are unable to figure out how to do so. 

![image](https://user-images.githubusercontent.com/92345662/211179079-2bbe0481-9e32-441b-baa4-7f4d03b4f571.png)

**### Describe the problem that you'd like solved**
Rectify issues with the creation of a user's LISTS.

1) Creating a reading list is not easy. It would be nice to be able to create a list directly from the " My Books" page.

Example:

![image](https://user-images.githubusercontent.com/92345662/211180819-bba49834-ee0d-465e-bcec-b280486949f5.png)

2) The drop down menu " WANT TO READ"  where the user can create a list should be the be the same drop down menu on books returned from the book search.

Example:

When searching for a book, this is the drop down menu the user sees:

![image](https://user-images.githubusercontent.com/92345662/211180975-16a8cd38-9974-4c91-996f-68f15312835a.png)

When clicking on a book, this is the drop down menu the user sees:

![image](https://user-images.githubusercontent.com/92345662/211181009-561934a3-2770-4d89-a97e-24250b62e319.png)

3)  " LISTS" should be named " My LISTS", to differentiate from public lists. 
Example of a public list.

![image](https://user-images.githubusercontent.com/92345662/211180566-2b07f4ab-ac86-4243-87d2-721cf8725e25.png)
 
These LISTS are also LISTS but they are not LISTS created by the user.

4)  " see all" should be switched to " edit" in all pages where user LIST appears. 

For example:

![image](https://user-images.githubusercontent.com/92345662/211181087-2ab713e8-42d5-4104-8823-15fd09164f5a.png)

5) Switching " See all" to "[edit]" will allow users to click on edit to : rename a list, change the list description, delete lists.

6) Deleting all books from a list should not delete the list. Book and Lists should be separate so that if user wants to keep the list but delete all the books from the list and add new books, the user can. 
Currently, if the user wants to delete the last book on the list, the user sees:

![image](https://user-images.githubusercontent.com/92345662/211181159-af517ea8-fa0d-4d35-b2d5-58fa6717e9ab.png)

**### Proposal & Constraints**
- [ ] #7613
2) Make all WANT TO READ drop down menus the same throughout the website. 
3) Rename LISTS to MY LISTS to differentiate user lists from public/ suggested lists.
4) Rename " See all" to [edit] in all pages where user MY LISTS appears. This will flow better with the site design and also provide users with a tool to edit their lists instead of just seeing user's lists.
5) Change " see all" to edit beside MY LISTS in all pages containing MY LISTS. User will click this button and be directed to a list of all their lists, The user can then change or delete any of their created lists.
6) Deleting all books from a user's list should not prompt the user to delete the entire list. User should have the option to keep an empty list.

**### Additional context**
These modifications are intended to enhance user engagement with Open Library and increase accessibility.

**Comment by carlsonrob** [comment]
I'll work on this.

**Comment by cdrini** [comment]
@DebbieSan , @JaydenTeoh is working on the first piece; Adding the "create a new list" button, which should just bring up the same popup. Can we review the mockup for this on Friday to get a 👍 from everyone so he is unblocked?

I split that out into a separate issue so I can assign it to him :)

**Comment by DebbieSan** [comment]
Hi @cdrini no problem :) , I will switch my focus to " My Books" so I can get a couple options before continuing with the dropper.

**Comment by cdrini** [comment]
Thanks!

**Comment by mekarpeles** [comment]
@DebbieSan @cdrini @jimchamp @szgrune --

I'm noticing we have a handful of issues around **List Improvements** and I consider the four of you stakeholders.

Can we meet to figure out how to turn this collection or proposals into a plan?

See also #3593. 

I'm proposing we ultimately close both this and #3593 in favor of an `epic` based on @DebbieSan's GSoC 2023 proposal here:
https://docs.google.com/document/d/1kThM4RDyLGnYKn2WsorxjiOybTajuHKNJJ5P9SLEvYc/edit

**Comment by mekarpeles** [comment]
Moving @spamspambaconspam's original comments here to close #3593: 

The ability for a user to create lists isn't really helpful, as currently implemented.

### Describe the problem that you'd like solved

- [x] Seeds API is broken: https://openlibrary.org/people/mekBot/lists/OL115530L/seeds.json

- [ ] Lists widget is unusably out of date and should have customizable embed options + styles: https://openlibrary.org/people/mekBot/lists/OL115530L/Staff_Picks/embed

- [ ] Add ability to sort items based on title, author, year, etc.
(Currently, the list is sorted by 'time since last edited'.  Not only is this sort almost useless, but if I edit a record that appears on someone else's list, my edit then modifies the sort of _their_ list.    If I 'fake edit' records to force a specific order on my list, I'm possibly screwing up someone else's previous 'fake edits' they'd painstakingly made to force their own list to be in a specific order.)

- [ ] Add ability to display or hide details of each record, in order to view more records in window.
(Here are some details that take up room but are not considered vital:

1. Displaying every single author.
2. Being able to download a DAISY file.
3. Displaying thumbnails.
4. Displaying date of last edit ~instead, the copyright date (for a work) or publication date (for an edition) would be more helpful.)

 
![image](https://user-images.githubusercontent.com/9367321/87798819-7a03dd80-c81a-11ea-9b80-40ee01bcd2ba.png)

**Comment by JaydenTeoh** [comment]
@mekarpeles Thanks for helping to gather the relevant issues into one major issue. This would really help with discussion and collaboration between the involved contributors. As for the issues mentioned in #3593, I believe the only issue that is still relevant would be the ability to sort lists. @DebbieSan would you mind giving me access to your GSoC proposal for the list design? I have requested permission for it. And we should definitely have a meeting to discuss the upcoming implementations for the List feature. What I am currently looking forward to is making List a community-driven feature and I have a few great ideas around it that I can't wait to share with you guys!

**Comment by mekarpeles** [comment]
* [ ] #6531
* [ ] #6992
* [ ] #7653

**Comment by DebbieSan** [comment]
Awesome. Thanks @mekarpeles I will work on this plan.

**Comment by DebbieSan** [comment]
Upon investigating, #7613 and #3070 seem to be addressing the same issue ( though the title made me think otherwise for a moment).

Both of these should be closed by #7553
